python/katana/lonestar/analytics/sssp.py

```diff
@@ -18,13 +18,25 @@
 from katana.galois import setActiveThreads
 
 
-UpdateRequest = np.dtype([("src", np.uint64), ("dist", np.uint32),])
+UpdateRequest = np.dtype(
+    [
+        ("src", np.uint64),
+        ("dist", np.uint32),
+    ]
+)
+
+
+def dtype_info(t):
+    t = np.dtype(t)
+    if t.kind == "f":
+        return np.finfo(t)
+    else:
+        return np.iinfo(t)
 
 
 def create_distance_array(g: PropertyGraph, source, length_property):
-    inf_distance = numba.types.uint64.maxval
     a = np.empty(len(g), dtype=dtype_of_pyarrow_array(g.get_edge_property(length_property)))
-    a[:] = inf_distance
+    a[:] = dtype_info(a.dtype).max
     a[source] = 0
```

Comment:
Just noting that `inf_distance `makes some intuitive sense where `dtype_info(a.dtype).max` does not.